From 84f46d6ad7981d10a3c598f19642b12e6036a4da Mon Sep 17 00:00:00 2001
From: Daiki Ueno <ueno@gnu.org>
Date: Fri, 9 Aug 2019 11:38:16 +0200
Subject: [PATCH] build: Declare creation methods of abstract classes as
 protected

Public creation methods of abstract classes are no longer allowed
since 0.45.1:
https://gitlab.gnome.org/GNOME/vala/commit/9365176e3f4fab737112e957f38c128752c8b504

Suggested by Rico Tzschichholz.
---
 libkkc/language-model.vala | 2 +-
 libkkc/metadata-file.vala  | 2 +-
 tests/lib/test-case.vala   | 2 +-
 3 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/libkkc/language-model.vala b/libkkc/language-model.vala
index 967746a..19860aa 100644
--- a/libkkc/language-model.vala
+++ b/libkkc/language-model.vala
@@ -101,7 +101,7 @@ namespace Kkc {
         public abstract new LanguageModelEntry? @get (string input,
                                                       string output);
 
-        public LanguageModel (LanguageModelMetadata metadata) throws Error {
+        protected LanguageModel (LanguageModelMetadata metadata) throws Error {
             Object (metadata: metadata);
             init (null);
         }
diff --git a/libkkc/metadata-file.vala b/libkkc/metadata-file.vala
index 62b84a3..f52235b 100644
--- a/libkkc/metadata-file.vala
+++ b/libkkc/metadata-file.vala
@@ -43,7 +43,7 @@ namespace Kkc {
          */
         public string filename { get; construct set; }
 
-        public MetadataFile (string name, string filename) throws Error {
+        protected MetadataFile (string name, string filename) throws Error {
             Object (name: name, filename: filename);
             init (null);
         }
diff --git a/tests/lib/test-case.vala b/tests/lib/test-case.vala
index da74f7e..116a57b 100644
--- a/tests/lib/test-case.vala
+++ b/tests/lib/test-case.vala
@@ -29,7 +29,7 @@ public abstract class Kkc.TestCase : Object
 
   public delegate void TestMethod ();
 
-  public TestCase (string name)
+  protected TestCase (string name)
     {
       this._suite = new GLib.TestSuite (name);
     }
-- 
2.21.0

